src/Microsoft.ApiDesignGuidelines.Analyzers/Core/InterfaceMethodsShouldBeCallableByChildTypes.cs

```diff
@@ -1,39 +1,142 @@
+                                                                          description: s_localizableDescription,
+                                                                          helpLinkUri: "https://msdn.microsoft.com/library/ms182153.aspx",
+                                                                          customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public sealed override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public sealed override void Initialize(AnalysisContext analysisContext) =>
+            analysisContext.RegisterOperationBlockAction(AnalyzeOperationBlock);
+
+        private bool ShouldExcludeOperationBlock(ImmutableArray<IOperation> operationBlocks)
+        {
+            if (operationBlocks != null && operationBlocks.Length == 1)
+            {
+                IBlockStatement block = operationBlocks[0] as IBlockStatement;
+
+                // An operation block that's not even a block - don't analyze the error cases.
+                if (block == null)
+                {
+                    return true;
+                }
+
+                if (block.Statements.Length == 0 ||
+                    (block.Statements.Length == 1 && block.Statements[0].Kind == OperationKind.ThrowStatement))
+                {
+                    // Empty body OR body that just throws.
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private void AnalyzeOperationBlock(OperationBlockAnalysisContext context)
+        {
+            if (context.OwningSymbol.Kind != SymbolKind.Method)
+            {
+                return;
+            }
+
+            var method = (IMethodSymbol)context.OwningSymbol;
+
+            // We are only interested in private explicit interface implementations within a public non-sealed type.
+            if (method.ExplicitInterfaceImplementations.Length == 0 ||
+                method.GetResultantVisibility() != SymbolVisibility.Private ||
+                method.ContainingType.IsSealed ||
+                method.ContainingType.GetResultantVisibility() != SymbolVisibility.Public)
+            {
+                return;
+            }
+
+            // Avoid false reports from simple explicit implementations where the deriving type is not expected to access the base implementation.
+            if (ShouldExcludeOperationBlock(context.OperationBlocks))
+            {
+                return;
+            }
+
+            var hasPublicInterfaceImplementation = false;
+            foreach (var interfaceMethod in method.ExplicitInterfaceImplementations)
+            {
+                // If any one of the explicitly implemented interface methods has a visible alternate, then effectively, they all do.
+                if (HasVisibleAlternate(method.ContainingType, interfaceMethod))
+                {
+                    return;
```

Comment:
The loop is enumerating the explicit implementations of the *method* and not the *type*. In C# methods cannot have multiple explicit implementation but in VB one can write this:
```vb.net
Sub Foo() Implements IFoo.Foo, IBar.Foo
```

Since this OperationAction is invoked once per method, your scenario will get covered anyway.